src/libertem/udf/com.py

```diff
@@ -0,0 +1,630 @@
+        By default, everything except the last row and last column are used
+        since these contain artefacts.
+
+    Returns
+    -------
+    (straight, flipped)
+        Root mean square of the curl as a function of :code:`scan_rotation` from
+        0 to 359 degrees in steps of one, with :code:`flip_y=False` (straight)
+        and code:`flip_y=True` (flipped).
+    '''
+    straight = np.zeros(360)
+    flipped = np.zeros(360)
+    if roi is None:
+        # The last row and column contain artifacts
+        roi = (slice(0, -1), slice(0, -1))
+    for angle in range(360):
+        for flip_y in (True, False):
+            y_transformed, x_transformed = apply_correction(
+                y_centers, x_centers, scan_rotation=angle, flip_y=flip_y
+            )
+            curl = curl_2d(y_transformed, x_transformed)
+            result = np.sqrt(np.mean(curl[roi]**2))
+            if flip_y:
+                flipped[angle] = result
+            else:
+                straight[angle] = result
+    return (straight, flipped)
+
+
+class GuessResult(NamedTuple):
+    scan_rotation: int
+    flip_y: bool
+    cy: float
+    cx: float
+
+
+def guess_corrections(
+    y_centers: np.ndarray,
+    x_centers: np.ndarray,
+    roi: Optional[Union[np.ndarray, Tuple[slice, ...]]] = None,
+) -> GuessResult:
+    '''
+    Guess corrections for center shift, :code:`scan_rotation` and :code:`flip_y` from CoM data
+
+    This function can generate a CoM parameter guess for atomic resolution 4D STEM data
+    by using the following assumptions:
+
+    * The field is purely electrostatic, i.e. the RMS curl should be minimized
+    * There is no net field over the field of view and no descan error,
+      i.e. the mean deflection is zero.
+    * Atomic resolution STEM means that the divergence will be negative at atom columns
+      and consequently the histogram of divergence will have a stronger tail towards
+      negative values than towards positive values.
+
+    If any corrections were applied when generating the input data, please note that the corrections
+    should be applied relative to these previous value. In particular, the
+    center corrections returned by this function have to be back-transformed to the uncorrected
+    coordinate system, for example with :code:`apply_correction(..., forward=False)`
+
+    Parameters
+    ----------
+    y_centers, x_centers : numpy.ndarray
+        2D arrays with y and x component of the center of mass shift for each
+        scan position, as returned by :meth:`center_shifts` or
+        :meth:`apply_correction`
+    roi : Optional[numpy.ndarray]
+        Selector for values to consider in the statistics, compatible
+        with indexing an array with the shape of y_centers and x_centers.
+        By default, everything except the last row and last column are used
+        since these contain artefacts.
+
+    Returns
+    -------
+    GuessResult : relative to current values
```

Comment:
Can we have an example here how to pipe the result from the `CoMUDF` into `guess_corrections`? I think that would be a common use-case.

That would also mean this function needs to be added to the reference docs, or it needs to be exposed (as a static method?) on the `CoMUDF` itself.

Reply:
Agreed for the example, even I am not 100% sure how to pipe from one to the other...

The `guess_corrections` function is already documented under `analysis.com`, though its docstring references the functions `center_shifts` and `apply_corrections` which are themselves undocumented, currently. I will see if I can write something complete with the new functionality offered with CoMUDF.

Reply:
Do you have a good way to :autodoc: a NamedTuple or Enum (e.g. https://libertem.github.io/LiberTEM/reference/api.html#libertem.analysis.com.GuessResult ) ? I think there are Sphinx extension that can do it, but otherwise the output is quite unhelpful!

Reply:
Yeah, named tuples aren't properly supported by sphinx, yet: https://github.com/sphinx-doc/sphinx/issues/8717

I wouldn't be opposed to using [`autonamedtuple`](https://sphinx-toolbox.readthedocs.io/en/latest/extensions/more_autodoc/autonamedtuple.html) from sphinx-toolbox, if that is compatible with the rest of our sphinx extensions.